### ISSUE TITLE
[Offcanvas-list] Parent Menu visible under Submenu

### ISSUE THREAD

**Issue by ArTiSTiX** [zurb/foundation-sites#6090]
I have an issue with submenu height and its parent menu height.

If parent menu is taller than submenu (so that user has to scroll on parent), then submenu is inside parent "viewport". So when user scrolls on submenu (even if submenu has no eventual overflow-y content), the parent menu appears under submenu.

This is because parent has position: absolute, so has submenus (whose height inherit from parent "viewport" height,  top & left depends on parent, and not on page, so scrolling is still triggered on parent).
The only solution i see is to break submenus hierarchy, chain it, and use link as the toggle reference.
But it will break fallback compatibility, so if someone has an idea about ho to fix it easily...

Reproduce bug on doc:
Create many `<li>` in parent menu -> open submenu -> scroll -> parent menu is visible under submenu.

![foundation-offcanvas-issue](https://cloud.githubusercontent.com/assets/1231011/5338535/797574da-7ed5-11e4-9204-193a32d60933.png)

Tested on Chrome 39, Firefox 33.

**Comment by navidkashani** [comment]
i have same issue... any update?

**Comment by socketz** [comment]
Can you post the code? Probably has incorrect architecture

**Comment by rafibomb** [comment]
Ya, I was able to replicate this. @ArTiSTiX @navidkashani Did you look into what's causing it? Have a solution?

**Comment by ArTiSTiX** [comment]
Any update or fix planned ?

As i said, The issue is due to nested ```position: absolute;```.
With a hierarchical markup, submenus inherit parent menu height. Thus, scrolling both scrolls submenu and parent menu, and leads to unwanted display.

The solution is to avoid nesting submenus, and put them side by side. If you implement submenus this way, we must update JS to use the link href  (or a data-submenu attributes) to reference the submenu id.

The markup could look like this:

```
<ul class="off-canvas-list">
  <li class="has-submenu"><a href="#my_submenu">To Submenu</a><li>
</ul>
<ul class="off-canvas-list left-submenu">
  <li><a href="#anything">Submenu item</a><li>
</ul>
```

To keep the original behaviour, we could do something like:

* on submenu link click:
** if '<li>' parent has class "has-submenu" and contains a submenu (original behavior)
*** open it
** else check if link has an id reference, and references a offcanvas-list with class submenu (left or right)
*** open it
** else
*** follow link (ignore has-submenu class).

I think "offcanvas-list left-submenu" classes are already well styled. It only lacks the javascript to open it ("move-right" if it's "left-submenu") with a reference.

**Comment by rafibomb** [comment]
@ArTiSTiX Looks like you have the right idea for a fix. Can you submit a PR for this or do you need any more guidance for it?

**Comment by mtpultz** [comment]
I'm still experiencing this using build 5.5.3 has this moved beyond development? Or is there anything missing from the Foundation 5 docs?

**Comment by zurbchris** [comment]
Sorry folks, we got a little carried away trying to tackle Foundation 6 that 5 was a little neglected.

@ArTiSTiX While this might work pretty nicely if implemented, it would also be completely breaking, and in the name of SemVer, won't happen until at least Foundation 7.

I can tell you that the scrolling to reveal a parent menu problem was solved in 6. As for solving it in F5, I will put a request to the community to solve it in a non-breaking manner. PR's requested!
